### ISSUE TITLE
🐞 Tally is not showing all of the contests

### ISSUE THREAD

**Issue by SteveMaier-IRT** [microsoft/electionguard-python#764]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Defect ID : 15953
ElectionGuard Tally only reports the last instance of contests with the same name

**Description**
Summary: ElectionGuard Tally only reports the last instance of contests with the same name 
Description:  The Microsoft-provided ElectionGuard Tally system does not handle multiple contests with the same name. It omits results from all except the last instance of those contests. 
Requirements: All contests must be displayed on the Tally 
Release: ElectionGuard 
Configuration: EGAdmin tablet provided by Microsoft 
Election: County test election based on the available information 
Regression: No 
Legacy: No 
Reproducible: Yes 
Pre-Conditions: One or more contests with the same name on the ballot 
Steps to Reproduce: 
Load the election 
Scan the fully marked ballots 
Close polls 
Take the EGDrive to the EGAdmin tablet and generate a Tally 
Expected Results: All contests on the ballot are listed on the Tally 
Actual Results: This single ballot covers multiple legislative districts and includes multiple contests with the same names, 
e.g., State Senator; State Representative, Position A; State Representative, Position B 
The Scan Tally includes each instance of those contests and all of their choices. 
The EGAdmin Tally only includes the last instances of those contests and only includes the choices associated with the last instances of those contests. 
Log Details: N/A 
Impact: Very high - Tally is missing election results 
Analysis/Additional Notes/Legacy Information: 
This test election was constructed as much as possible to be an appropriate test election based on the information provided to QA about the pilot customer. 
Not yet known if the limitation applies to individual ballots or to the whole election. 
In other words, if no one ballot includes repeated contest names, can the Tally handle that? 
The QA test election does not include ballots with no repeated contest names. 
Defect Data: N/A 

**Defect Details**
Defect ID	15953
Status	New
Project	Scan
Subject	ElectionGuard
Reproducible	Y
Severity	4-Very High
Priority	3-High
Detected on Date	2022-08-19
Defect Type	Product Defect
Modified	2022-08-19 16:10:17
Detected in Release	ElectionGuard
Detected in Cycle	Drop 4
Target Release	ElectionGuard

### Environment

```markdown
- OS:
```
